### ISSUE TITLE
Compiling CODA fails with Python support

### ISSUE THREAD

**Issue by PHedelt** [stcorp/coda#52]
When compiling and making CODA with Python support, I get the following error message when running the make install command:
make  install-am
make[1]: Entering directory '/data/hedelt/coda'
w119,451 -python -Ipython -I./python -DPRINTF_ATTR=  -o python/codac.c ./python/codac.i
make[1]: w119,451: Command not found
make[1]: [Makefile:3413: python/codac.c] Error 127 (ignored)

I have pulled the latest version of CODA today...

**Comment by svniemeijer** [comment]
You need to make sure that you have swig installed. Otherwise you won't be able to generate the python interface.

**Comment by PHedelt** [comment]
Thanks for your quick response…
I have installed SWIG now in my HOME directory, but the same error appears… Do I need to provide the SWIG path to CODA somehow???

**Comment by svniemeijer** [comment]
You need to make sure that everything is properly rechecked by the configure script.
Run: `./config.status --recheck`.
Make sure that in the output where it says `checking for swig...` it points to your swig version.

If not, then you will have the run the configure script from scratch and you then can pass `SWIG=$HOME/<pathtoswig>/swig` as parameter to the configure script.

**Comment by PHedelt** [comment]
Ok, I had to add it to my PATH variable and rerun configure and make…
Now it compiled without problems.

However, a new issue appears:
When I am importing it now into python, I get the following error message:

import coda
ImportError: /users/hedelt/coda/lib64/python3.6/site-packages/coda/_codac.so: undefined symbol: coda_get_libcoda_version

**Comment by svniemeijer** [comment]
Loading _codac.so requires the coda.so library (from `/users/hedelt/coda/lib`) to be on your library search path. You probably need to set LD_LIBRARY_PATH.

**Comment by PHedelt** [comment]
My LD_LIBRARY_PATH variable points to /users/hedelt/coda/lib64/ but the error still appears...

**Comment by svniemeijer** [comment]
Is that really where your `coda.so` library is? Usually the python library path is `lib64` but the general library path is `lib`.
You can verify it by running `ldd /users/hedelt/coda/lib64/python3.6/site-packages/coda/_codac.so`. It should give you the fill path to the coda.so library.

**Comment by PHedelt** [comment]
Yes, in /users/hedelt/coda/lib64/ there are libcoda.so.15.0.2 and libcoda.so.15.0.1 etc. There is no lib path in the coda directory…

Nevertheless, using the ldd command, libcoda.so points to a different location… libcoda.so.14 => /data/upas2-resources/inst/lib64/libcoda.so.14

Why is that???

**Comment by svniemeijer** [comment]
I am not sure, but that is likely the reason of your problem.

**Comment by PHedelt** [comment]
I added the above library path for the ./configure command to point to the HDF5 and HDF4 libraries, which are located in /data/upas2-resources/inst/lib64/, so somehow CODA apparently found an older libcoda there... Is there a way to tell him to use the libcoda.so in my directory?

**Comment by svniemeijer** [comment]
I am not sure. And I don't think it will work at runtime (since still need `/data/upas2-resources/inst/lib64/` in your runtime path because of the HDF libraries).
You should either use `/data/upas2-resources/inst` as your installation path or you use a separate installation of HDF4/HDF5 (in a location that does not have a different coda version installed).

**Comment by PHedelt** [comment]
The solution to this problem was to link CODA with the system internal HDF libraries and not the libraries in upas2-resources. Somehow this caused CODA to link to an old libcoda in this directory...

This can be closed now - many thanks for your feedback!

**Comment by PHedelt** [comment]
The solution to this problem was to link CODA with the system internal HDF libraries and not the libraries in upas2-resources. Somehow this caused CODA to link to an old libcoda in this directory...

This can be closed now - many thanks for your feedback!
